### PR TITLE
Date used to calculate credit claim - percolating up

### PR DESCRIPTION
**Pull request by lmcewanhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/models/CreditsAnswer.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import uk.gov.hmrc.plasticpackagingtaxreturns.models.cache.UserAnswers
+
+// TODO add test coverage
+
+case class CreditsAnswer(yesNo: Boolean, weight: Option[Long]) {
+  def value: Long = (yesNo, weight) match {
+    case (true, Some(x)) => x
+    case (true, None) => 0
+    case (false, _) => 0
+  }
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/models/CreditsAnswer.scala:29]:
```suggestion
  def value: Long = (yesNo, weight) match {
    case (true, Some(x)) => x
    case _ => 0
  }
```

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/models/cache/UserAnswers.scala [hunk 2]
```diff
@@ -52,12 +51,35 @@ final case class UserAnswers(
...
+
+  /**
+   * Quickly set lots of (top-level) fields. Can be either 
+   * @param all varargs of (key: String, value: A) or (key: String, value: JsValue)
+   * @param writes [[Writes]] for A
+   * @tparam A allows values without eg JsNumber wrapper, if all values are same type
+   * @return [[UserAnswers]] with previous answer merged with given key-value
+   * @see UserAnswerSpec
+   */
   def setAll[A](all: (String, A)*) (implicit writes: Writes[A]): UserAnswers = {
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/models/cache/UserAnswers.scala:63]: these methods, and below are for test only?

**Review by liampan — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/models/CreditsAnswer.scala (hunk 1 above)
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/models/CreditsAnswer.scala:29]: nice

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/models/cache/UserAnswers.scala (hunk 2 above)
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtaxreturns/models/cache/UserAnswers.scala:63]: not necessarily, but only used by tests at the mo

**Pull request merged**